dev/release/verify-release-candidate.sh

```diff
@@ -202,6 +202,7 @@ test_and_install_cpp() {
 
   ARROW_CMAKE_OPTIONS="
 ${ARROW_CMAKE_OPTIONS}
+-DARROW_DEPENDENCY_SOURCE=BUNDLED
```

Comment:
Not sure it's ok to include this here.

Reply:
What error do you get when using conda for dependencies?

Reply:
When running the release script, I don't use conda for dependencies. AFAIU it should check downloading those.

Reply:
@xhochy

Reply:
I think `bundled` is the closest one to the previous behaviour.

Reply:
I think that `-DARROW_DEPENDENCY_SOURCE=AUTO` is better.
Anyway, please remove this change from this pull request.
It's not related to "[C++] Fix compilation warnings in release mode". I'll create a pull request for this case.

Reply:
Done.